File: 剑指offer/python/8，跳台阶.py
"""
跳台阶，本质上是费波切纳数列
"""


def jump_floor(number):
    a, b = 1, 2
    if number == 0:
        return 0
    for i in range(1,number):
        a, b = b, a + b

    return a


for i in range(10):
    print(jump_floor(i))

File: cpp_version/剑指offer/8，跳台阶.cpp
//
// Created by Jiang,Xinyang on 2020/7/28.
//

#include <iostream>

using namespace std;


int jumpFloor(int number) {
    if (number == 0)
        return 0;
    int a = 1;
    int b = 2;

    for (int i = 1; i < number; i++){
        int temp = a;
        a = b;
        b = temp + b;
    }
    return a;
}


int main() {

    int res = jumpFloor(9);
    cout << res << endl;

}